Update Hive ACID table seems sometime to corrupt the data

for example `TestHiveTransactionalTable.testAcidUpdateSelectedValues` is flaky https://github.com/trinodb/trino/actions/runs/3203025750/jobs/5233758071

```
2022-10-07T11:11:00.0565869Z tests               | 2022-10-07 16:56:00 INFO: FAILURE     /    io.trino.tests.product.hive.TestHiveTransactionalTable.testAcidUpdateSelectedValues (Groups: hive_transactional) took 2.8 seconds
2022-10-07T11:11:00.0566612Z tests               | 2022-10-07 16:56:00 SEVERE: Failure cause:
2022-10-07T11:11:00.0567331Z tests               | java.lang.AssertionError: [onTrino] Expected row count to be <2>, but was <4>; rows=[[7, ONE, 1000, true, 101], [13, DEUX, 2235, false, 202], [13, DEUX, 2235, false, 202], [7, ONE, 1000, true, 101]]
2022-10-07T11:11:00.0568062Z tests               | 	at io.trino.tests.product.hive.TestHiveTransactionalTable.verifySelect(TestHiveTransactionalTable.java:2230)
2022-10-07T11:11:00.0568883Z tests               | 	at io.trino.tests.product.hive.TestHiveTransactionalTable.verifySelectForTrinoAndHive(TestHiveTransactionalTable.java:2220)
2022-10-07T11:11:00.0569731Z tests               | 	at io.trino.tests.product.hive.TestHiveTransactionalTable.lambda$testAcidUpdateSelectedValues$57(TestHiveTransactionalTable.java:1309)
2022-10-07T11:11:00.0570515Z tests               | 	at io.trino.tests.product.hive.TestHiveTransactionalTable.withTemporaryTable(TestHiveTransactionalTable.java:1810)
2022-10-07T11:11:00.0571360Z tests               | 	at io.trino.tests.product.hive.TestHiveTransactionalTable.testAcidUpdateSelectedValues(TestHiveTransactionalTable.java:1302)
2022-10-07T11:11:00.0572092Z tests               | 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2022-10-07T11:11:00.0572734Z tests               | 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
2022-10-07T11:11:00.0573420Z tests               | 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2022-10-07T11:11:00.0573909Z tests               | 	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
2022-10-07T11:11:00.0574488Z tests               | 	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
2022-10-07T11:11:00.0575111Z tests               | 	at org.testng.internal.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:54)
2022-10-07T11:11:00.0575570Z tests               | 	at org.testng.internal.InvokeMethodRunnable.run(InvokeMethodRunnable.java:44)
2022-10-07T11:11:00.0576136Z tests               | 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
2022-10-07T11:11:00.0576680Z tests               | 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2022-10-07T11:11:00.0577245Z tests               | 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
2022-10-07T11:11:00.0577736Z tests               | 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
2022-10-07T11:11:00.0578358Z tests               | 	at java.base/java.lang.Thread.run(Thread.java:833)
```

cc @djsstarburst

Hmm, I had not looked into this before.  Summarizing:

Here is the exception:

```
2022-10-07T11:11:00.0567331Z tests               | java.lang.AssertionError: [onTrino] Expected row count to be <2>, but was <4>; rows=[[7, ONE, 1000, true, 101], [13, DEUX, 2235, false, 202], [13, DEUX, 2235, false, 202], [7, ONE, 1000, true, 101]
```

Here is what the test is doing:

```
CREATE TABLE tablename (col1 TINYINT, col2 VARCHAR, col3 BIGINT, col4 BOOLEAN, col5 INT);
INSERT INTO tablename (col1, col2, col3, col4, col5) VALUES (7, 'ONE', 1000, true, 101), (13, 'TWO', 2000, false, 202);

UPDATE tablename SET col2 = 'DEUX', col3 = col3 + 20 + col1 + col5 WHERE col1 = 13;

SELECT * FROM tablename;

Result should be:
VALUES (7, 'ONE', 1000, true, 101),
       (13, 'DEUX', 2235, false, 202)

Result was:       
VALUES (7, 'ONE', 1000, true, 101),
       (7, 'ONE', 1000, true, 101),
       (13, 'DEUX', 2235, false, 202),
       (13, 'DEUX', 2235, false, 202);
```

The UPDATE operation should change a single row.  However, in the following SELECT operation, both rows in the result set are duplicated.

--- a/testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveTransactionalTable.java
+++ b/testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveTransactionalTable.java
@@ -1299,7 +1299,7 @@ public class TestHiveTransactionalTable
     @Test(groups = HIVE_TRANSACTIONAL, timeOut = TEST_TIMEOUT)
     public void testAcidUpdateSelectedValues()
     {
-        withTemporaryTable("acid_update_simple", true, true, NONE, tableName -> {
+        withTemporaryTable("acid_update_simple_selected", true, true, NONE, tableName -> {
             onTrino().executeQuery(format("CREATE TABLE %s (col1 TINYINT, col2 VARCHAR, col3 BIGINT, col4 BOOLEAN, col5 INT) WITH (transactional = true)", tableName));
             log.info("About to insert");
             onTrino().executeQuery(format("INSERT INTO %s (col1, col2, col3, col4, col5) VALUES (7, 'ONE', 1000, true, 101), (13, 'TWO', 2000, false, 202)", tableName));
@@ -1806,7 +1806,7 @@ public class TestHiveTransactionalTable
         if (transactional) {
             ensureTransactionalHive();
         }
-        try (TemporaryHiveTable table = TemporaryHiveTable.temporaryHiveTable(tableName(rootName, isPartitioned, bucketingType))) {
+        try (TemporaryHiveTable table = TemporaryHiveTable.temporaryHiveTable(tableName(rootName, isPartitioned, bucketingType) + randomTableSuffix())) {
             testRunner.accept(table.getName());
         }
     }

